src/samVisualize/rendering/icons.ts

```diff
@@ -0,0 +1,37 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { readdirSync } from 'fs'
+import { join } from 'path'
+import * as vscode from 'vscode'
+/**
+ * Maps a resource type to a Webview URI pointing to a corresponding svg icon
+ */
+export type IconURIMap = {
+    [index: string]: string
+}
+/**
+ * Creates a map between a resource type and a webview URI pointing to a corresponding svg icon
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-template-resource-type-ref.html
+ * @param iconDir The path to the directory holding icons
+ * @param webview The webview from which URIs pointing to svg icons is produced
+ * @returns an IconURIMap mapping each svg icon in `iconDir`
+ */
+export function generateIconsMap(iconDir: string, webview: vscode.Webview): IconURIMap {
+    const iconsMap: IconURIMap = {}
+    const icons = readdirSync(iconDir)
+
+    for (const icon of icons) {
+        const vscfile = vscode.Uri.file(join(iconDir, icon))
+        // Convert the icon name from <a>-<b>-<c>.svg to <a>::<b>::<c>
+        // to match with Cloudformation resource names, which will be used as keys to retrieve the paths.
```

Comment:
> to match with CloudFormation

Aren't these all lowercase?

Reply:
Yeah, right now the icon names are all lowercase as well, but we shouldn't need to depend on that. I'll add a `toLowerCase()` so that the casing of the icon names doesn't matter.

Reply:
Okay, might want to also clarify the comment, as it won't strictly match CloudFormation.

Reply:
Yep, fixed.